main/export-list.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const ipc = require('electron-better-ipc');
+const base64Img = require('base64-img');
+const tmp = require('tmp');
+const ffmpeg = require('@ffmpeg-installer/ffmpeg');
+const util = require('electron-util');
+const execa = require('execa');
+
+const {showExportsWindow, getExportsWindow, openExportsWindow} = require('./exports');
+const {openEditorWindow} = require('./editor');
+const Export = require('./export');
+
+const ffmpegPath = util.fixPathForAsarUnpack(ffmpeg.path);
+
+const getPreview = async inputPath => {
+  const previewPath = tmp.tmpNameSync({postfix: '.jpg'});
+  await execa(ffmpegPath, [
+    '-ss', 0,
+    '-i', inputPath,
+    '-t', 1,
+    '-vframes', 1,
+    '-f', 'image2',
+    previewPath
+  ]);
+
+  return base64Img.base64Sync(previewPath);
+};
+
+const saveSnapshot = async ({inputPath, outputPath, time}) => {
+  execa(ffmpegPath, [
+    '-i', inputPath,
+    '-ss', time,
+    '-vframes', 1,
+    outputPath
+  ]);
+};
+
+class ExportList {
+  constructor() {
+    this.exports = [];
+    this.queue = [];
+  }
+
+  async _startNext() {
+    if (this.queue.length === 0) {
+      return;
+    }
+
+    this.currentExport = this.queue.shift();
+    if (this.currentExport.canceled) {
+      this._startNext();
+      return;
+    }
+
+    this.currentExport.run()
+      .then(() => {
```

Comment:
Hmm, after reading this part of the code again, I don't see what this should be `await`ed. The method just starts the export. If we use `await` it will wait for it to be done running, right?